Example/AppDelegate.swift

```diff
@@ -21,7 +21,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
   private func setupNetworkActivityHandler(for application: UIApplication) {
     NetworkServiceActivity.activityChangeHandler = { isActive in
-      application.isNetworkActivityIndicatorVisible = isActive
+      DispatchQueue.main.async {
```

Comment:
Ooh that thread sanitiser catches so much. Love it.

Reply:
It's awesome! I was thinking that we probably should call this block on the main thread within the library, thoughts?

Reply:
It's awesome! I was thinking that we probably should call this block on the main thread within the library, thoughts?

Reply:
Would it be possible to let people choose what queue to dispatch on, defaulting to `main`?

Reply:
Yeah you're right actually. Stick with main!